compiler/il/OMRBlock.cpp

```diff
@@ -1245,42 +1245,47 @@ static TR::SymbolReference * createSymRefForNode(TR::Compilation *comp, TR::Reso
  *       register. It also makes sure that regStore tree is moved to before split point also marks used register unavailable.
  *    c. If node to be uncommoned is stored into a register before split point and node does not succeeds in [b] case and register is available then replacement
  *       node will be regLoad from that register also marks used register unavailable.
- * 3. While checking for case-b and case-c from Step 2 it also checks regStores and PassThrough nodes after split point. 
+ * 3. While checking for case-b and case-c from Step 2 it also checks regStores and PassThrough nodes after split point.
  *    a. If node under pass through requires to be uncommoned and uses different register then the replacement node
  *       then it stores replacement node into a register used by passThrough before the tree containing that passthrough node.
- *    b. If node under regStore does not need to be uncommoned, then mark the used register unavailable. 
+ *    b. If node under regStore does not need to be uncommoned, then mark the used register unavailable.
  * 4. For the remaining of nodes to be uncommoned, it tries to find the available register using list of unavailable registers and if it can find one, then stores node
- *    into that register otherwise creates a auto/temp slot to uncommon them. 
- * 5. Last step is to start walking treetops after split point and replace nodes to be uncommoned with corresponding replacement nodes.
- * 
+ *    into that register otherwise creates a auto/temp slot to uncommon them.
+ * 5. Start walking treetops after split point and replace nodes to be uncommoned with corresponding replacement nodes.
+ * 6. If _liveLocals exists and isCopyLiveLocals is true, copy _liveLocals from the current block to the new block.
+ *
  * @param startOfNewBlock TreeTop from which new block will start
  * @param cfg TR::CFG object
  * @param copyExceptionSuccessors Boolean stating if we need to copy the exceptionSuccessors of the original blocks to new block
  * @param methodSymbol TR::ResolvedMethodSymbol object
+ * @param fixupCommoning Boolean stating if we need to uncommon nodes between blocks
+ * @param isCopyLiveLocals If we need to copy _liveLocals from the current block to the new block if _liveLocals exists
  * @return TR::Block* Returns new Block
  */
 TR::Block *
-OMR::Block::splitPostGRA(TR::TreeTop * startOfNewBlock, TR::CFG *cfg, bool copyExceptionSuccessors, TR::ResolvedMethodSymbol *methodSymbol)
+OMR::Block::splitPostGRA(TR::TreeTop * startOfNewBlock, TR::CFG *cfg, bool copyExceptionSuccessors, TR::ResolvedMethodSymbol *methodSymbol,
+                         bool fixupCommoning, bool isCopyLiveLocals)
    {
    TR::Compilation *comp = cfg->comp();
    TR::Block *newBlock = self()->split(startOfNewBlock, cfg, false, copyExceptionSuccessors, methodSymbol);
    if (methodSymbol == NULL)
       methodSymbol = comp->getMethodSymbol();
 
 
+   if (fixupCommoning)
```

Comment:
@andrewcraik  To make reviewing easier, this is the main change. All the steps from identifying node which are referenced after split point as well to replacing of them is done if `fixUpCommoning` is set to true. Otherwise, it will just copy information of live locals if it is available to newly created block. Changes in rest of the places, I believe is cleaning up extra spaces.

Reply:
That's correct and the functional change also includes  `// Step - 6  : Copy _liveLocals from the current block to the new block if _liveLocals exists.`. The rest of changes just cleans up trailing whitespace as Rahil pointed out.